python/lsst/obs/fiberspectrograph/translator.py

```diff
@@ -0,0 +1,214 @@
+        # guess
+        if self.is_key_ok("EXPTIME"):
+            return self.quantity_from_card("EXPTIME", u.s)
+
+        # A missing or undefined EXPTIME is problematic. Set to -1
+        # to indicate that none was found.
+        log.warning("%s: Insufficient information to derive exposure time. Setting to -1.0s",
+                    self._log_prefix)
+        return -1.0 * u.s
+
+    @cache_translation
+    def to_dark_time(self):             # N.b. defining this suppresses a warning re setting from exptime
+        if "DARKTIME" in self._header:
+            darkTime = self._header["DARKTIME"]
+            self._used_these_cards("DARKTIME")
+            return (darkTime, dict(unit=u.s))
+        return self.to_exposure_time()
+
+    @staticmethod
+    def compute_exposure_id(dayobs, seqnum, controller=None):
+        """Helper method to calculate the exposure_id.
+
+        Parameters
+        ----------
+        dayobs : `str`
+            Day of observation in either YYYYMMDD or YYYY-MM-DD format.
+            If the string looks like ISO format it will be truncated before the
+            ``T`` before being handled.
+        seqnum : `int` or `str`
+            Sequence number.
+        controller : `str`, optional
+            Controller to use. If this is "O", no change is made to the
+            exposure ID. If it is "C" a 1000 is added to the year component
+            of the exposure ID. If it is "H" a 2000 is added to the year
+            component. This sequence continues with "P" and "Q" controllers.
+            `None` indicates that the controller is not relevant to the
+            exposure ID calculation (generally this is the case for test
+            stand data).
+
+        Returns
+        -------
+        exposure_id : `int`
+            Exposure ID in form YYYYMMDDnnnnn form.
+        """
+        if not isinstance(dayobs, int):
+            if "T" in dayobs:
+                dayobs = dayobs[:dayobs.find("T")]
+
+            dayobs = dayobs.replace("-", "")
+
+            if len(dayobs) != 8:
+                raise ValueError(f"Malformed dayobs: {dayobs}")
+
+        # Expect no more than 99,999 exposures in a day
+        maxdigits = 5
+        if seqnum >= 10**maxdigits:
+            raise ValueError(f"Sequence number ({seqnum}) exceeds limit")
+
+        # Form the number as a string zero padding the sequence number
+        idstr = f"{dayobs}{seqnum:0{maxdigits}d}"
+
+        # Exposure ID has to be an integer
+        return int(idstr)
+
+    @cache_translation
+    def to_visit_id(self):
+        """Calculate the visit associated with this exposure.
+        """
+        return None
+
+    @cache_translation
+    def to_exposure_id(self):
+        """Generate a unique exposure ID number
+
+        This is a combination of DAYOBS and SEQNUM
+
+        Returns
+        -------
+        exposure_id : `int`
+            Unique exposure number.
+        """
+        if "CALIB_ID" in self._header:
```

Comment:
This code is not needed. You aren't trying to read calibration files as if they are raws.